### ISSUE TITLE
Filter on Span Kind

### ISSUE THREAD

**Issue by tbbream** [open-telemetry/opentelemetry-collector-contrib#12615]
**Is your feature request related to a problem? Please describe.**
Filter on span kind

**Describe the solution you'd like**
Only capture spans for certain kind, (like consumer and server). This way, can get spanmetrics for just the entry points to each task.

**Describe alternatives you've considered**
Tried to use transform to add an attribute by matching on kind, but it doesn't appear to work on matching against kind.

**Additional context**

**Comment by TylerHelmuth** [comment]
@tbbream When using the transform processor did you interact with kind using an int? The field is returned as an int64 bc it is an Enum. 

With 0.56.0 that was released today you can now interact with the field via the Enum symbol from the traces protobuf instead of an int if you'd like.

**Comment by TylerHelmuth** [comment]
What processor were you hoping to use to filter? The filter processor? Or where you hoping to configure the spanmetrics processor to only generate metrics for certain span kinds?

**Comment by tbbream** [comment]
I was thinking that the spanmetrics would not need to change, but using a filter in front of it to reduce the number of traces going into that processor.

**Comment by tbbream** [comment]
@TylerHelmuth I was trying to use transform to set an attribute matching on the kind. I was using the integers. Here is the config I was using:
```
    receivers:
      otlp:
        protocols:
          grpc:
          http:
      otlp/spanmetrics:
        protocols:
          grpc:
            endpoint: "localhost:12345"
  
    processors:
      batch:
      transform:
        traces:
          queries:
            - set(attributes["process.kind"], kind)
            - set(attributes["process.status"], status.code)
            - set(attributes["capture.kind"], "unspecified") where kind == 0
            - set(attributes["capture.kind"], "internal")  where kind == 1
            - set(attributes["capture.kind"], "server") where kind == 2
            - set(attributes["capture.kind"], "client") where kind == 3
            - set(attributes["capture.kind"], "producer") where kind == 4
            - set(attributes["capture.kind"], "consumer") where kind == 5
            - set(attributes["capture.kind"], "nil") where kind == nil
            - set(attributes["transform"], "happened")
      filter:
        spans:
          include:
            match_type: strict
            attributes:
              - Key: capture.spm
                Value: yes
      spanmetrics:
        metrics_exporter: otlp/spanmetrics
        latency_histogram_buckets: [1ms, 2ms, 5ms, 10ms, 100ms, 500ms, 1s, 5s, 10s, 15s]
        dimensions:
          - name: host.name
          - name: k8s.pod.name
          - name: http.status
          - name: track.spm
    
    exporters:
      logging:
        loglevel: debug
    
      prometheus:
        endpoint: localhost:8889
        metric_expiration: 5m
    
      otlp/spanmetrics:
        endpoint: "localhost:4317"
        tls:
          insecure: true
    
    service:
      pipelines:
        traces:
          receivers: [otlp]
          processors: [transform,spanmetrics]
          exporters: [logging]
    
        metrics/spanmetrics:
          # This receiver is just a dummy and never used.
          # Added to pass validation requiring at least one receiver in a pipeline.
          receivers: [otlp/spanmetrics]
          exporters: [otlp/spanmetrics]
    
        metrics:
          receivers: [otlp]
          exporters: [prometheus]
```

Here is the outputted trace.
```
2022-07-21T13:02:51.142Z	DEBUG	loggingexporter/logging_exporter.go:52	ResourceSpans #0
Resource SchemaURL:
Resource labels:
     -> telemetry.sdk.language: STRING(python)
     -> telemetry.sdk.name: STRING(opentelemetry)
     -> telemetry.sdk.version: STRING(1.11.1)
     -> host.name: STRING(localhost)
     -> k8s.namespace.name: STRING(local)
     -> k8s.pod.ip: STRING(localhost)
     -> k8s.pod.name: STRING(local)
     -> service.name: STRING(testing)
     -> service.version: STRING(1.1.0)
ScopeSpans #0
ScopeSpans SchemaURL:
InstrumentationScope test
Span #0
    Trace ID       : e616f4024db74eb22c155932d3ab6df2
    Parent ID      :
    ID             : e8dc15c43171083a
    Name           : hello
    Kind           : SPAN_KIND_CONSUMER
    Start time     : 2022-07-21 13:02:49.711568 +0000 UTC
    End time       : 2022-07-21 13:02:49.711649 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message :
Attributes:
     -> track.spm: STRING(no)
     -> transform: STRING(happened)
```

**Comment by TylerHelmuth** [comment]
@tbbream what version of the collector did you use?

**Comment by tbbream** [comment]
@TylerHelmuth `image: ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector-contrib:0.55.0`

**Comment by TylerHelmuth** [comment]
I got ahead of myself,  the [defect with accessing `kind` in a where clause](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10349) wasn't fixed until 0.56.0.  [Once this issue is closed](https://github.com/open-telemetry/opentelemetry-collector-releases/issues/166), you should be able to use 0.56.0 and your setup should work.

You'll also be able to switch from ints to [enum symbols](  enum SpanKind {) if you like (`... where kind == SPAN_KIND_UNSPECIFIED`)

**Comment by tbbream** [comment]
Just tried on the new version and the transform is working to assign an attribute matching on a span kind. This meets my needs and I am closing this.
